instaloader/instaloader.py

```diff
@@ -103,22 +103,16 @@ def format_field(self, value, format_spec):
 
 
 class _PostPathFormatter(_ArbitraryItemFormatter):
-    def vformat(self, format_string, args, kwargs):
-        """Override :meth:`string.Formatter.vformat` for character substitution in paths for Windows, see issue #84."""
-        ret = super().vformat(format_string, args, kwargs)
-        if platform.system() == 'Windows':
-            ret = ret.replace(':', '\ua789').replace('<', '\ufe64').replace('>', '\ufe65').replace('\"', '\uff02')
-            ret = ret.replace('\\', '\ufe68').replace('|', '\uff5c').replace('?', '\ufe16').replace('*', '\uff0a')
-        return ret
-
     def get_value(self, key, args, kwargs):
-        """Replaces '/' with similar looking Division Slash and on windows newline with space"""
+        """Replaces '/' with similar looking Division Slash. Also replaces illegal filename characters on windows."""
         ret = super().get_value(key, args, kwargs)
         if not isinstance(ret, str):
```

Comment:
There are two cases where the _PostPathFormatter is invoked with a `Path` as target:
1. `target=Path(profile.username) / Path(':tagged')` https://github.com/instaloader/instaloader/blob/bf2f4f16ed7648fb5970a3d6c0d3e23152c0a6fd/instaloader/instaloader.py#L944-L946
2. `target=Path(name) / Path(user_highlight.title)` https://github.com/instaloader/instaloader/blob/bf2f4f16ed7648fb5970a3d6c0d3e23152c0a6fd/instaloader/instaloader.py#L682-L684

With get_value() only doing the replacement logic when it gets a `str`, these cases are not covered, so we'd miss the colon replacement in the first case, and the character replacements in the highlight title in the second case.

This could be fixed by invoking the replacement logic when `ret` is a `str`, *and* for each Path part when `ret` is a `Path` – Only for each part of the Path, so that we do not replace the path separator (slash / backslash) between the username and `:tagged` or highlight title while still replacing title characters that are disallowed in Windows paths.

Reply:
Resolved in 852b0eec2134b2a1b0129dc53be189b06a6dcbfb